### PR TITLE
Move parameter validation inside conditional

### LINKED ISSUE #358: foreman::plugin::discovery fails without tftp installed on the foreman server
Because foreman::plugin::discovery validates $tftp_root_clean even when $install_images is false, it is impossible to install the plugin on foreman servers with external tftp servers.

### PR DESCRIPTION
**Pull request by brandonweeks**
Fixes #358

### CODE REVIEW

**Comment by mmoll** [comment]
I didn't test this, but it's looking good :+1:

File: manifests/plugin/discovery.pp [hunk 1]
```diff
@@ -21,16 +21,16 @@
...
-  validate_absolute_path($tftp_root_clean)
-  validate_string($source_url)
-  validate_string($image_name)
-
   foreman::plugin {'discovery':
   }
 
   if $install_images {
+    $tftp_root_clean = regsubst($tftp_root, '/$', '')
+    validate_bool($install_images)
```
Comment (ekohl) [inline manifests/plugin/discovery.pp]: I think can stay outside the if statement since it's used there.

**Comment by brandonweeks** [comment]
@ekohl corrected.

**Comment by mmoll** [comment]
merged, thanks @brandonweeks!

**Pull request closed without merging**